packages/core/content-releases/admin/src/components/CMReleasesContainer.tsx

```diff
@@ -0,0 +1,280 @@
+import * as React from 'react';
+
+import {
+  Box,
+  Button,
+  FieldLabel,
+  Flex,
+  ModalBody,
+  ModalHeader,
+  ModalLayout,
+  SingleSelect,
+  SingleSelectOption,
+  Typography,
+  ModalFooter,
+} from '@strapi/design-system';
+import {
+  CheckPermissions,
+  useAPIErrorHandler,
+  useCMEditViewDataManager,
+  useNotification,
+} from '@strapi/helper-plugin';
+import { Plus } from '@strapi/icons';
+import { isAxiosError } from 'axios';
+import { Formik, Form } from 'formik';
+import { useIntl } from 'react-intl';
+import { useParams } from 'react-router-dom';
+import * as yup from 'yup';
+
+import { CreateReleaseAction } from '../../../shared/contracts/release-actions';
+import { PERMISSIONS } from '../constants';
+import { useCreateReleaseActionMutation, useGetReleasesQuery } from '../services/release';
+
+import { ReleaseActionOptions } from './ReleaseActionOptions';
+
+const RELEASE_ACTION_FORM_SCHEMA = yup.object().shape({
+  type: yup.string().oneOf(['publish', 'unpublish']).required(),
+  releaseId: yup.string().required(),
+});
+
+interface FormValues {
+  type: CreateReleaseAction.Request['body']['type'];
+  releaseId: CreateReleaseAction.Request['params']['releaseId'];
+}
+
+const INITIAL_VALUES = {
+  type: 'publish',
+  releaseId: '',
+} satisfies FormValues;
+
+interface AddActionToReleaseModalProps {
+  handleClose: () => void;
+}
+
+const AddActionToReleaseModal = ({ handleClose }: AddActionToReleaseModalProps) => {
+  const { formatMessage } = useIntl();
+  const toggleNotification = useNotification();
+  const { formatAPIError } = useAPIErrorHandler();
+  const params = useParams<{ id?: string }>();
+  const {
+    allLayoutData: { contentType },
+  } = useCMEditViewDataManager();
+  // Get all 'pending' releases
+  const response = useGetReleasesQuery({
+    filters: {
+      $and: [
+        {
+          releasedAt: {
+            $null: true,
+          },
+        },
+      ],
+    },
+  });
```

Comment:
Maybe this is something you want to handle when showing the associated releases but here we should exclude the releases already assigned to this entry

Reply:
I think we will need to handle this in another PR. It's a UX thing but definitely needed. I created a task in Jira with highest priority. I'll mention this query needs to be updated.